src/copyright/agent/database.cc

```diff
@@ -300,21 +300,38 @@ bool CopyrightDatabaseHandler::insertNoResultInDatabase(long int agentId, long i
 
 bool CopyrightDatabaseHandler::insertInDatabase(DatabaseEntry& entry) const
 {
-  return dbManager.execPrepared(
-    fo_dbManager_PrepareStamement(
-      dbManager.getStruct_dbManager(),
-      "insertInDatabase",
-      "INSERT INTO "
-      IDENTITY
-      "(agent_fk, pfile_fk, content, hash, type, copy_startbyte, copy_endbyte)"
-        " VALUES($1,$2,$3,md5($3),$4,$5,$6)",
-      long, long, char*, char*, int, int
-    ),
-    entry.agent_fk, entry.pfile_fk,
-    entry.content.c_str(),
-    entry.type.c_str(),
-    entry.copy_startbyte, entry.copy_endbyte
-  );
+  if(!strcmp(entry.type.c_str(),IDENTITY) || strcmp(entry.type.c_str(),"statement")){
```

Comment:
Currently the if statement does not work for `ecc` since if `entry.type.c_str() == "ecc" == IDENTITY` the statment
```
!strcmp(entry.type.c_str(),IDENTITY) || strcmp(entry.type.c_str(),"statement"
```
evaluates to true and writes `ecc` information to the `author` table.